universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -329,8 +328,6 @@ private void putParentArray(Path path) {
       // For n>0, only add the nth element if the n-1 element exists.
     } else if (hasPath(parentPath.atIndex(index - 1))) {
       addAt(parentPath, new HashMap<>());
-
-      // TODO(#624): remove this recursion.
```

Comment:
TODO is outdated as we chose to take a different design direction: https://github.com/seattle-uat/civiform/issues/624#issuecomment-845325307